libraries/wutdevoptab/devoptab_fsa_utils.cpp

```diff
@@ -29,14 +107,38 @@ __wut_fsa_fixpath(struct _reent *r,
       return NULL;
    }
 
-   fixedPath = static_cast<char *>(memalign(0x40, FS_MAX_PATH + 1));
+   uint32_t maxPathLength = FS_MAX_PATH + 1;
+   fixedPath = static_cast<char *>(memalign(0x40, maxPathLength));
    if (!fixedPath) {
       r->_errno = ENOMEM;
       return NULL;
    }
 
-   // cwd is handled by coreinit, so just strip the 'device:' if it exists
-   strcpy(fixedPath, p);
+   if (canonicalizedPath) {
+      // Convert to an absolute path
+      if (p[0] != '\0' && p[0] != '\\' && p[0] != '/') {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-truncation"
+         __wut_fsa_device_t *deviceData = (__wut_fsa_device_t *) r->deviceData;
+         snprintf(fixedPath, maxPathLength, "%s/%s", deviceData->cwd, p);
```

Comment:
same thing here

         if (snprintf(fixedPath, maxPathLength, "%s/%s", deviceData->cwd, p) >= maxPathLength)
            fixedPath[maxPathLength] = '\0';

but also change the type of maxPathLength to int.